src/H5Omessage.c

```diff
@@ -499,6 +499,11 @@ H5O_msg_read_oh(H5F_t *f, H5O_t *oh, unsigned type_id, void *mesg)
     type = H5O_msg_class_g[type_id]; /* map the type ID to the actual type object */
     HDassert(type);
 
+    if ((oh->nmesgs * sizeof(H5O_mesg_t)) > oh->cache_info.size) {
```

Comment:
There was an issue with how I was executing the test suite before.  This isn't correct - it throws a false positive for most HDF5 files.